components/scream/src/physics/shoc/shoc_functions_f90.hpp

```diff
@@ -42,6 +43,11 @@ struct SHOCDataBase
     for (size_t i = 0; i < m_ptrs_i.size(); ++i) {
       util::transpose<D>(*(m_ptrs_i[i]), data.data() + (m_ptrs.size()*m_total) + (m_totali*i), shcol, nlevi);
     }
+    
+    // Transpose on the column only grid
+    for (size_t i = 0; i < m_ptrs_c.size(); ++i) {
+      util::transpose<D>(*(m_ptrs_c[i]), data.data() + (m_ptrs.size()*m_total) + (m_ptrs_i.size()*m_totali) + (m_totalc*i), shcol, 1);
```

Comment:
Single dimension data does not need to be transposed.

Reply:
Thanks for looking at this.  I understand the points above and below this one, but am struggling with this one.  Do you imply that I can delete this loop?  If so then all tests that contain column only arrays fail.  I'm guessing I need to change something else to compensate for this change and have toyed around a bit but can't seem to get it to work.  any pointers?

Reply:
Ah, because of
```
m_data = data
```
below. A straight copy of 1-dimensional is needed:

```
const c_start_offset = m_ptrs.size()*m_total + m_ptrs_i.size()*m_totali;
std::copy(m_data.begin() + c_start_offset, m_data.end(), data.begin() + c_start_offset); 
```

Reply:
Thanks Jim.  I assume I need to do a "const auto c_start_offset" or "const Int c_start_offset" (else compiler complains about explicit type missing)?  In any case, if I declare a type I get another error which I attach a screenshot.  I tried to do a google search of what this error means but all it did was drive me into full panic mode about how much I have to learn about c++ 
![Screen Shot 2020-08-20 at 12 02 18 PM](https://user-images.githubusercontent.com/25110300/90813900-1e2ae980-e2dd-11ea-817a-76c226abb8d4.png)

Reply:
@bogensch , apologies, I messed up my code same. c_start_offset should be a `const Int`. In your code, it looks like you have an `=` where you want a `+`.

Reply:
doh!  thanks!